src/functions/template-tags/general.php

```diff
@@ -5,2032 +5,1733 @@
+	if ( ! is_numeric( $post ) && ! $post instanceof WP_Post ) {
+		$post = get_the_ID();
 	}
 
-	/**
-	 * Return the views registered in the tribe bar
-	 *
-	 * @category Events
-	 *
-	 * @return array
-	 **/
-	function tribe_events_get_views() {
-		return apply_filters( 'tribe-events-bar-views', [] );
+	// If not a WP_Post we try to fetch it as one
+	if ( is_numeric( $post ) ) {
+		$post = WP_Post::get_instance( $post );
 	}
 
-	/**
-	 * Returns the URL for use in the tribe bar form's action attribute.
-	 *
-	 * @return string URL for current tribe bar form action.
-	 */
-	function tribe_events_get_current_filter_url() {
-		global $wp;
+	// Prevent Non usable $post instances
+	if ( ! $post instanceof WP_Post ) {
+		return null;
+	}
 
-		$url = esc_url( add_query_arg( $wp->query_string, '', home_url( $wp->request ) ) );
-		return apply_filters( 'tribe_events_get_current_filter_url', $url );
+	// Default Allowed HTML
+	if ( ! is_array( $allowed_html ) ) {
+		$base_attrs   = [
+			'class' => [],
+			'id'    => [],
+			'style' => [],
+		];
+		$allowed_html = [
+			'a'      => [
+				'class'  => [],
+				'id'     => [],
+				'style'  => [],
+				'href'   => [],
+				'rel'    => [],
+				'target' => [],
+			],
+			'b'      => $base_attrs,
+			'strong' => $base_attrs,
+			'em'     => $base_attrs,
+			'span'   => $base_attrs,
+			'ul'     => $base_attrs,
+			'li'     => $base_attrs,
+			'ol'     => $base_attrs,
+		];
 	}
 
+	$post_id = $post->ID;
+
 	/**
-	 * Get and increment tab index in form fields
+	 * Allow developers to filter what are the allowed HTML on the Excerpt
+	 *
+	 * @link https://codex.wordpress.org/Function_Reference/wp_kses
+	 *
+	 * @param WP_Post $post The current post object.
 	 *
+	 * @param array Must be compatible to wp_kses structure.
 	 */
-	function tribe_events_get_tab_index() {
-		$tribe_events = Tribe__Events__Main::instance();
-
-		return apply_filters( 'tribe_events_tab_index', $tribe_events->tabIndex() );
-	}
+	$allowed_html = apply_filters( 'tribe_events_excerpt_allowed_html', $allowed_html, $post );
 
 	/**
-	 * Echo and increment tab index in form fields
+	 * Allow shortcodes to be Applied on the Excerpt or not
 	 *
+	 * @param bool Whether shortcodes are allowed in the Event excerpt or not.
+	 * @param WP_Post $post The current post object.
 	 */
-	function tribe_events_tab_index() {
-		echo tribe_events_get_tab_index();
-	}
+	$allow_shortcodes = apply_filters( 'tribe_events_excerpt_allow_shortcode', false, $post );
 
 	/**
-	 * Check if a particular view is enabled
+	 * Filter to stop removal of shortcode markup in the excerpt.
 	 *
-	 * @category Events
+	 * This will remove all text that resembles a shortcode [shortcode 5].
 	 *
-	 * @param string $view Name of view to check, should match what's in Tribe__Events__Main->displaying when on that view
+	 * @since 5.1.0
 	 *
-	 * @return bool
-	 **/
-	function tribe_events_is_view_enabled( $view ) {
```

Comment:
I brought this back and [refactored it](https://github.com/the-events-calendar/the-events-calendar/pull/3788/files#diff-d405ece6fce146a7eb1fc2f822e4d10626724fb92519bdd94881a48d65684752R1313) to use Views\V2\Manager.